packages/react-next/src/components/FocusTrapZone/FocusTrapZone.tsx

```diff
@@ -6,131 +6,97 @@ import {
-      !elementContains(this._root.current, this._getDocument().activeElement as HTMLElement)
-    ) {
-      this._returnFocusToInitiator();
-    }
-
-    // Dispose of event handlers so their closures can be garbage-collected
-    if (this._disposeClickHandler) {
-      this._disposeClickHandler();
-      this._disposeClickHandler = undefined;
-    }
-
-    if (this._disposeFocusHandler) {
-      this._disposeFocusHandler();
-      this._disposeFocusHandler = undefined;
-    }
+import { IFocusTrapZoneProps, IFocusTrapZone } from './FocusTrapZone.types';
+import { useId, useConst, useMergedRefs } from '@uifabric/react-hooks';
+import { useDocument } from '@fluentui/react-window-provider';
+
+interface IFocusTrapZoneInternalState {
+  disposeFocusHandler: (() => void) | undefined;
+  disposeClickHandler: (() => void) | undefined;
+  previouslyFocusedElementOutsideTrapZone: HTMLElement | undefined;
+  previouslyFocusedElementInTrapZone: HTMLElement | undefined;
+  hasFocus: boolean;
+}
 
-    // Dispose of element references so the DOM Nodes can be garbage-collected
-    delete this._previouslyFocusedElementInTrapZone;
-    delete this._previouslyFocusedElementOutsideTrapZone;
-  }
-
-  public render(): JSX.Element {
-    const { className, disabled = false, ariaLabelledBy } = this.props;
-    const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties);
-
-    const bumperProps = {
-      'aria-hidden': true,
-      style: {
-        pointerEvents: 'none',
-        position: 'fixed', // 'fixed' prevents browsers from scrolling to bumpers when viewport does not contain them
+const COMPONENT_NAME = 'FocusTrapZone';
+
+const useComponentRef = (
+  componentRef: IRefObject<IFocusTrapZone> | undefined,
+  previouslyFocusedElement: HTMLElement | undefined,
+  focus: () => void,
+) => {
+  React.useImperativeHandle(
+    componentRef,
+    () => ({
+      get previouslyFocusedElement() {
+        return previouslyFocusedElement;
       },
-      tabIndex: disabled ? -1 : 0, // make bumpers tabbable only when enabled
-      'data-is-visible': true,
-    } as React.HTMLAttributes<HTMLDivElement>;
-
-    return (
-      <div
-        {...divProps}
-        className={className}
-        ref={this._root}
-        aria-labelledby={ariaLabelledBy}
-        onFocusCapture={this._onFocusCapture}
-        onFocus={this._onRootFocus}
-        onBlur={this._onRootBlur}
-      >
-        <div {...bumperProps} ref={this._firstBumper} onFocus={this._onFirstBumperFocus} />
-        {this.props.children}
-        <div {...bumperProps} ref={this._lastBumper} onFocus={this._onLastBumperFocus} />
-      </div>
-    );
-  }
-
-  public focus() {
-    const { focusPreviouslyFocusedInnerElement, firstFocusableSelector } = this.props;
-
+      focus,
+    }),
+    [previouslyFocusedElement, focus],
+  );
+};
+
+export const FocusTrapZone: React.FunctionComponent<IFocusTrapZoneProps> & {
+  focusStack: string[];
```

Comment:
this should not be here. we should just define `focusStack` in file scope

Reply:
focusStack is a static. Minimally it needs to be file scoped, but the tests also need access to it, which is why it's attached as a static.

Reply:
should we avoid adding this to public api then? we can still access it the same way in our tests, just without the typing:
```
export const FocusTrapZone: React.FunctionComponent<IFocusTrapZoneProps> & {
   focusStack: string[];
};
```

Reply:
we can take this as follow up and address in future PR if needed

Reply:
We looked at removing, but I don't think it's worth addressing. Recommend we leave as is.